packages/langchain-graph-retriever/src/langchain_graph_retriever/adapters/astra.py

```diff
@@ -226,3 +228,84 @@ async def _aget_by_id_with_embedding(self, document_id: str) -> Content | None:
             projection=self.vector_store.document_codec.full_projection,
         )
         return self._hit_to_content(hit)
+
+
+    @override
+    def get_adjacent(
+        self,
+        edges: set[Edge],
+        query_embedding: list[float],
+        k: int,
+        filter: dict[str, Any] | None,
+        **kwargs: Any,
+    ) -> Iterable[Content]:
+        filter_values: dict[str, Any] = {}
+        for edge in edges:
+            if isinstance(edge, MetadataEdge):
+                filter_values.setdefault(
+                    f"metadata.{edge.incoming_field}", []).append(edge.value)
+            elif isinstance(edge, IdEdge):
+                filter_values.setdefault("_id", []).append(edge.id)
+
+        ids = filter_values.pop("_id", [])
```

Comment:
I initially tried doing this all in one query, but Astra rejects `_id` filters within an `"$or"`, so I needed to do it as two queries and then combine them.